recipes/conversational_voice_agent/app.py

```diff
@@ -146,16 +153,44 @@ def chat(history: List) -> List[List[str]]:
 
 def synthesize(conversation: List[List[str]]) -> Tuple[int, np.ndarray]:
     """
-    Generate audio from text
+    Synthesizes speech from the last message in a conversation using a TTS model.
+
+    Parameters:
+        conversation (List[List[str]]): A list of message pairs, each pair containing a
+                                        user prompt and an assistant response.
 
-    Params:
-        conversation: conversation history with the chatbot
     Returns:
-        Sample rate and generated audio in form of numpy array
+        Tuple[int, np.ndarray]: A tuple containing the sampling rate (int) and the
+                                synthesized audio as a numpy ndarray.
     """
+    start_time = time.time()  # Start time
     prompt = conversation[-1][-1]
-    # generate audio for the last prompt (bot's response)
-    return SAMPLE_RATE, tts_model.generate_audio(prompt)
+
+    # Function to split text into sentences
+    def split_into_sentences(text):
+        sentences = re.split(r'(?<=[^A-Z].[.!?]) +(?=[A-Z])', text)
+        return [sentence.strip() for sentence in sentences if sentence.strip()]
+
+    # Split the prompt into sentences
+    sentences = split_into_sentences(prompt)
+    audio_segments = []
+
+    for sentence in sentences:
+        inputs = tts_processor(text=sentence, return_tensors="pt")
+
+        # Check if the token count is within the limit
+        if inputs.input_ids.size(1) <= 600:
```

Comment:
What happens if the input is longer?

Reply:
I will address by implementing the chunk_sentence function. This function breaks down sentences that exceed the token limit into smaller parts. Each chunk is then separately processed by the TTS model.